cpu/atmega_common/periph/uart.c

```diff
@@ -83,9 +85,16 @@ static uart_isr_ctx_t isr_ctx[UART_NUMOF];
 
 static void _update_brr(uart_t uart, uint16_t brr, bool double_speed)
 {
+#ifndef CPU_ATMEGA8
     dev[uart]->BRR = brr;
+#else /* on atmega8 BRRH is shared with CSRC */
+    dev[uart]->CSRC = (brr >> 8);
+    dev[uart]->BRRL = (uint8_t)(brr & 0x00ff);
+#endif
     if (double_speed) {
-#ifdef CPU_ATMEGA32U4
+#if defined(CPU_ATMEGA8)
+        dev[uart]->CSRA |= (1 << U2X);
+#elif defined(CPU_ATMEGA32U4)
```

Comment:
Always feature not CPU

You can create a bit fixup inside driver

```diff
+#ifndef U2X1
+  #ifdef U2X
+    #define U2X1 U2X
+  #endif
+#endif
```

Then simple use it

```diff
+#ifdef U2X1
-#ifdef CPU_ATMEGA32U4
    dev[uart]->CSRA |= (1 << U2X1);
#else
...
``````

Reply:
I prefer to keep all the bit defines in avr/io.h